netket/hilbert/fock.py

```diff
@@ -152,7 +158,7 @@ def __repr__(self):
             if self._n_particles is not None
             else ""
         )
-        nmax = self._n_max if self._n_max < np.iinfo(np.intp).max else "INT_MAX"
+        nmax = self._n_max if self._n_max < FOCK_MAX else "INT_MAX"
```

Comment:
Maybe this should say FOCK_MAX now?
```suggestion
        nmax = self._n_max if self._n_max < FOCK_MAX else "FOCK_MAX"
```